### ISSUE TITLE
Windows builds broken further recently

### ISSUE THREAD

**Issue by greerga** [Perl/perl5#17597]
One of the recent changes for "mbtowc" functions broke Windows further.  For reference, the last good commit tested was:

011c085a2f90    Karl Williamson: Update to latest Unicode 13.0

At the time, the Windows 10 build result was F and Windows 2000 was FM.

The first bad commit tested was:

5a6637f01ac3    Karl Williamson: Fixup POSIX::mbtowc, wctomb

Afterward the Windows 10 build result is FM and Windows 2000 is M.

I can restore Windows 2000 building with this change:

```
diff --git a/perlvars.h b/perlvars.h
index 5a351a0..1a9259e 100644
--- a/perlvars.h
+++ b/perlvars.h
@@ -107,9 +107,9 @@ PERLVAR(G, hints_mutex, perl_mutex)    /* Mutex for refcounted he refcounti
 #  if ! defined(USE_THREAD_SAFE_LOCALE) || defined(TS_W32_BROKEN_LOCALECONV)
 PERLVAR(G, locale_mutex, perl_mutex)   /* Mutex for setlocale() changing */
 #  endif
-#  ifndef USE_THREAD_SAFE_LOCALE
+/* #  ifndef USE_THREAD_SAFE_LOCALE */
 PERLVAR(G, lc_numeric_mutex, perl_mutex)   /* Mutex for switching LC_NUMERIC */
-#  endif
+/* #  endif */
 #endif

 #ifdef USE_POSIX_2008_LOCALE
```

but I'm not confident that is the best change to truly adjust the issues.

**Comment by xenu** [comment]
What is your `CCTYPE` set to?

**Comment by Leont** [comment]
> but I'm not confident that is the best change to truly adjust the issues.

I'm pretty sure it isn't.

I would guess `LOCALE_LOCK` should be a no-op on platforms where no such lock is necessary.

**Comment by khwilliamson** [comment]
On 3/1/20 8:13 AM, Leon Timmermans wrote:
>     but I'm not confident that is the best change to truly adjust the
>     issues.
> 
> I'm pretty sure it isn't.
> 
> I would guess |LOCALE_LOCK| should be a no-op on platforms where no such 
> lock is necessary.
> 
That's the way it should already be.

**Comment by gannett-ggreer** [comment]
CCTYPE on my Windows 2000 smoker is MSVC80FREE and the Windows 10 smoker is MSVC142.

**Comment by xenu** [comment]
>CCTYPE on my Windows 2000 smoker is MSVC80FREE 

Changing it to `MSVC80` will slightly unbreak your build. `*FREE` CCTYPES were removed by 3aa3d69ac5e208c16829ec576c5f16f90681d60c

**Comment by khwilliamson** [comment]
On 3/1/20 2:52 PM, xenu wrote:
>     CCTYPE on my Windows 2000 smoker is MSVC80FREE
> 
> Changing it to |MSVC80| will slightly unbreak your build. |*FREE| 
> CCTYPES were removed by 3aa3d69 
> <https://github.com/Perl/perl5/commit/3aa3d69ac5e208c16829ec576c5f16f90681d60c>
> 
> —

Indeed, when I tried it on a windows box, I got the PL_lc_numeric_mutex 
error, but when I removed FREE, it compile and passed its tests, threaded.

**Comment by gannett-ggreer** [comment]
It would have been nice for the Makefile to error out at the unrecognized, formerly-valid value.  I didn't realize it was gone.

That doesn't explain the Windows 10 failures though, like: http://perl.develop-help.com/raw/?id=248277

**Comment by khwilliamson** [comment]
@steve-m-hay   In makedef.pl, we could check for FREE and raise a warning or error?  Or is this a one-off thing that won't likely happen again.

I don't have access to a Win 10 box to debug on.

**Comment by khwilliamson** [comment]
Note also that this is passing on blead in a win 10 smoke situation:
Sample pass: https://github.com/Perl/perl5/actions/runs/47827116

What is different in @gannett-ggreer  configuration from this one?

**Comment by gannett-ggreer** [comment]
The GitHub workflow sets environment variables that I do not:
>        PERL_SKIP_TTY_TEST: 1
>        CONTINUOUS_INTEGRATION: 1

(Test::Smoke may set the former itself under harness.)

I call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvars64.bat" but that should be functionally equivalent to the GitHub workflow because that batch file is:
```
 @call "%~dp0vcvarsall.bat" x64 %*
```
Any other differences would involve Test::Smoke setup.

**Comment by tonycoz** [comment]
I believe this was fixed by c83b64ee3d2f2626c7c692d434d7e1659b2c05d6.

Can you confirm @greerga ?

There's a similar issue #17601 which tracks a different problem.

**Comment by greerga** [comment]
Neither the Windows 10 nor Windows 2000 smoker have an M status on any stage now, so I'd call that fixed.
